src/plone/restapi/services/users/add.py

```diff
@@ -62,11 +83,27 @@ def validate_input_data(self, portal, original_data):
         # check input data
         for fieldname in required:
             if not data.get(fieldname, None):
-                self.add_field_error(fieldname, f"Property '{fieldname}' is required.")
+                translated_fieldname = self.translate_fieldname(fieldname)
+                self.add_field_error(
+                    fieldname,
+                    self.translate(
+                        _(
+                            "Property '${fieldname}' is required.",
+                            mapping={"fieldname": translated_fieldname},
+                        )
+                    ),
+                )
         for fieldname in data:
             if fieldname not in allowed:
+                translated_fieldname = self.translate_fieldname(fieldname)
                 self.add_field_error(
-                    fieldname, f"Property '{fieldname}' is not allowed."
+                    fieldname,
+                    self.translate(
+                        _(
+                            "Property '${fieldname}' is not allowed.",
+                            mapping={"fieldname": translated_fieldname},
+                        )
+                    ),
```

Comment:
Since the basic structure is repeated here twice, that makes me think that maybe the translation logic should go inside `add_field_error` to avoid repeating. This is just an idea and not critical, though.

Reply:
Now I call the translation inside the _error and add_field_error methods.